frontend/kesaseteli/employer/browser-tests/application-page/step2.components.ts

```diff
@@ -30,124 +30,179 @@ export const getStep2Components = (t: TestController) => {
     });
   const withinForm = (): ReturnType<typeof within> => within(formSelector());
 
+  const accordionSelector = (isOpen = true, index = 0) =>
+    screen.findByTestId(`accordion-${index}-${isOpen ? 'open' : 'closed'}`);
+  const withinAccordion = (
+    isOpen = true,
+    index = 0
+  ): ReturnType<typeof within> => within(accordionSelector(isOpen, index));
+
+  const form = async () => {
+    const selectors = {
+      title() {
+        return formSelector();
+      },
+    };
+    const expectations = {
+      async isPresent() {
+        await t
+          .expect(selectors.title().exists)
+          .ok(await getErrorMessage(t), { timeout: 10_000 });
+      },
+    };
+    const actions = {
+      async openAllClosedAccordions() {
+        /* eslint-disable no-await-in-loop */
+        for (;;) {
+          const closedAccordion = accordionSelector(false);
+          if (await closedAccordion.exists) {
+            await t.click(closedAccordion);
+          } else {
+            break;
+          }
+        }
+        /* eslint-enable no-await-in-loop */
+      },
+      async removeAllAccordions() {
+        /* eslint-disable no-await-in-loop */
+        for (;;) {
+          const removeButtons = within(formSelector()).findAllByRole('button', {
+            name: /poista tiedot/i,
+          });
+          if (await removeButtons.exists) {
+            await t.click(removeButtons.nth(0));
+          } else {
+            break;
+          }
+        }
+        /* eslint-enable no-await-in-loop */
+      },
+    };
+    await expectations.isPresent();
+    return {
+      expectations,
+      actions,
+    };
+  };
```

Comment:
Although these functions seem to do the job, they might be error-prone for tests.

Reply:
It was only way I was able to work with it. There might be draft applications from previous runs which must be removed before starting to fill a new one.. each application has remove button so there might be 0...n remove buttons on page. Each one should be clicked until there is no more remove buttons on page. Also SelectorPromise is not a list, only way to access nth button is to use `removeButtons.nth(0)`. Also every click affects to dom, so you have to re-valuate the selector after each remove... I mean that after `t.click(removeButtons.nth(0))` is done, then `t.click(removeButtons.nth(1))` would not work after it but we have to re-evaluate the `removeButtons` selector in between and slways take the `nth(0)` from the list. 
Anyway it seem to work very well. I tried to cut the previous run in different situations, also manually added 8 applications, and also tried taht there is zero draft applications. In each case it worked correctly.